src/Hl7.Fhir.Base/Utility/StringExtensions.cs

```diff
@@ -139,14 +138,30 @@ internal static string[] SplitNotEscaped(this string value, char separator)
         /// <param name="text"></param>
         /// <param name="prefix"></param>
         /// <returns></returns>
-        public static bool MatchesPrefix(this string text, string prefix)
+        public static bool MatchesPrefix(this ReadOnlySpan<char> text, ReadOnlySpan<char> prefix)
         {
-            var prefixMatch = prefix?.EndsWith("*") ?? false;
+            bool wildcard = prefix.EndsWith("*".AsSpan()); //used for value[x] elements
```

Comment:
Since we are optimizing anyway, shall we not make a 

`static readonly Span Asterix = "*".AsSpan()`

That you can use in the `EndsWith` and the `TrimEnd`, so we can re-use that single span instead of creating two of them in every call?